### PR TITLE
WT-8704 Tidy up CMake warnings on Windows builds

### PR DESCRIPTION
**Pull request by alisonfel**
When using newer versions of CMake & VS MSBuild, we get a significant number of warnings at both the CMake configuration stages and during an `MSBuild` compilation. The PR intends to tidy these warnings up, the key changes being:
- Added `CMAKE_MODULE_LINKER_FLAGS` to build modes: This being a missing flag variable that is appended to a given CMake build mode. Newer versions of CMake will flag a warning about the missing variable, this change addresses the warning.
- Defined `None` build mode: The `None` build mode is the default CMake build mode, where no additional compiler flags are appended. We want to explicitly define it as newer versions of CMake will display warning messages about the None build mode not existing in the CMake cache. This change shouldn't have an effect on the build, we still get the same functionality with or without the line. This change is just intended to suppress spurious warning messages.
- Moved select Windows warning diagnostics to main config file: The MSVC warnings `C4090` & `C4996` are quite prominent irrespective of a strict or non-strict build. As we don't intend to fix these issues, moved these diagnostic warnings to the main Windows config file to reduce noise.

### CODE REVIEW

File: cmake/configs/modes.cmake [hunk 1]
```diff
@@ -95,11 +95,16 @@ function(define_build_mode mode)
         "${linker_flags}" CACHE STRING
         "Linker lags to be used to create shared libraries for ${mode} build type." FORCE)
 
+    set(CMAKE_MODULE_LINKER_FLAGS_${build_mode}
+        "${linker_flags}" CACHE STRING
+        "Linker lags to be used to create shared modules for ${mode} build type." FORCE)
```
Comment (korteland) [inline cmake/configs/modes.cmake:100]: nitpick: "lags" -> "flags"

**Review by korteland — CHANGES_REQUESTED** [review]
LGTM modulo the spelling error :-)

**Review by etienneptl — APPROVED** [review]
LGTM

File: cmake/configs/modes.cmake (hunk 1 above)
Comment (alisonfel) [reply cmake/configs/modes.cmake]: Oh whoops, good find. Fixed in 367aa6f

**Review by korteland — APPROVED** [review]
LGTM, thanks!

**Pull request merged**